src/StreamWrapper.php

```diff
@@ -106,30 +125,28 @@ public function stream_metadata(string $path, int $option, mixed $value): bool
 
     public function stream_open(string $path, string $mode, int $options, ?string &$opened_path): bool
     {
-        $fd = substr($path, strlen('inotify://'));
-        if (!is_numeric($fd)) {
-            throw new Exception(sprintf(
-                'Invalid file descriptor in "%s": "%s"',
-                $path,
-                $fd,
-            ));
+        $fd = \substr($path, \strlen('inotify://'));
+        if (!\is_numeric($fd)) {
+            \trigger_error(\sprintf('Invalid file descriptor in "%s": "%s"', $path, $fd), E_USER_WARNING);
```

Comment:
I actually liked throwing an exception here

Reply:
By default, opening an invalid resource does not throw an exception, fopen function just return false with warning in this case. Don't you think it would be better to be closer to default behavior here?